packages/builders/src/components/ActionRow.ts

```diff
@@ -98,38 +85,164 @@ export class ActionRowBuilder<ComponentType extends AnyComponentBuilder> extends
+	public constructor({ components = [], ...data }: Partial<APIActionRowComponent<APIActionRowComponentTypes>> = {}) {
+		super();
+		this.data = {
+			...structuredClone(data),
+			type: ComponentType.ActionRow,
+			components: components.map((component) => createComponentBuilder(component)) ?? [],
+		};
 	}
 
 	/**
-	 * Adds components to this action row.
+	 * Adds custom id button components to this action row.
 	 *
-	 * @param components - The components to add
+	 * @param input - The buttons to add
 	 */
-	public addComponents(...components: RestOrArray<ComponentType>) {
-		this.components.push(...normalizeArray(components));
+	public addCustomIdButtonComponents(
+		...input: RestOrArray<
+			| APIButtonComponentWithCustomId
+			| CustomIdButtonBuilder
+			| ((builder: CustomIdButtonBuilder) => CustomIdButtonBuilder)
+		>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((component) => resolveBuilder(component, CustomIdButtonBuilder));
+
+		this.data.components.push(...resolved);
+		return this;
+	}
+
+	/**
+	 * Adds SKU id button components to this action row.
+	 *
+	 * @param input - The buttons to add
+	 */
+	public addSKUIDButtonComponents(
+		...input: RestOrArray<
+			APIButtonComponentWithSKUId | SKUIdButtonBuilder | ((builder: SKUIdButtonBuilder) => SKUIdButtonBuilder)
+		>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((component) => resolveBuilder(component, SKUIdButtonBuilder));
+
+		this.data.components.push(...resolved);
 		return this;
 	}
 
 	/**
-	 * Sets components for this action row.
+	 * Adds URL button components to this action row.
 	 *
-	 * @param components - The components to set
+	 * @param input - The buttons to add
 	 */
-	public setComponents(...components: RestOrArray<ComponentType>) {
-		this.components.splice(0, this.components.length, ...normalizeArray(components));
+	public addURLButtonComponents(
+		...input: RestOrArray<
+			APIButtonComponentWithURL | URLButtonBuilder | ((builder: URLButtonBuilder) => URLButtonBuilder)
+		>
+	): this {
+		const normalized = normalizeArray(input);
+		const resolved = normalized.map((component) => resolveBuilder(component, URLButtonBuilder));
+
+		this.data.components.push(...resolved);
+		return this;
+	}
+
+	/**
+	 * Adds a channel select menu component to this action row.
+	 *
+	 * @param input - A function that returns a component builder or an already built builder
+	 */
+	public addChannelSelectMenuComponent(
+		input:
+			| APIChannelSelectComponent
+			| ChannelSelectMenuBuilder
+			| ((builder: ChannelSelectMenuBuilder) => ChannelSelectMenuBuilder),
+	): this {
+		this.data.components.push(resolveBuilder(input, ChannelSelectMenuBuilder));
+		return this;
+	}
+
+	/**
+	 * Adds a mentionable select menu component to this action row.
+	 *
+	 * @param input - A function that returns a component builder or an already built builder
+	 */
+	public addMentionableSelectMenuComponent(
```

Comment:
Same as `ChatInputCommand`. Should we add a way to clear components after adding them?

Reply:
Here it's pretty important, yeah. We can make a splice method work.